src/design.jl

```diff
@@ -97,6 +97,40 @@ end
 """
 $(TYPEDSIGNATURES)
 
+```jldoctest
+julia> fold!(PlackettBurman(2))
+8×3 DataFrame
+│ Row │ factor1 │ factor2 │ dummy1 │
+│     │ Int64   │ Int64   │ Int64  │
+├─────┼─────────┼─────────┼────────┤
+│ 1   │ 1       │ 1       │ 1      │
+│ 2   │ 1       │ -1      │ -1     │
+│ 3   │ -1      │ -1      │ 1      │
+│ 4   │ -1      │ 1       │ -1     │
+│ 5   │ -1      │ -1      │ -1     │
+│ 6   │ -1      │ 1       │ 1      │
+│ 7   │ 1       │ 1       │ -1     │
+│ 8   │ 1       │ -1      │ 1      │
+
+```
+"""
+function fold!(design::AbstractDesign)
```

Comment:
The problem of targeting `AbstractDesign`s is that some symbols will not be `:low` or `:high`, and that many numerical values are not encoded as `-1` and `1`. Maybe this is a design flaw in this package for numerical factors, but I would argue that it is better to let the user decide on contrasts and encoding when fitting some model to the design.

Reply:
Yes, I agree, AbstractDesign is too abstract...